sql/tztime.cc

```diff
@@ -1169,19 +1170,10 @@ void Time_zone_utc::gmt_sec_to_TIME(MYSQL_TIME *tmp, my_time_t t) const {
   SYNOPSIS
     get_name()
 
-  DESCRIPTION
-    Since Time_zone_utc is used only internally by SQL's UTC_* functions it
-    is not accessible directly, and hence this function of Time_zone
-    interface is not implemented for this class and should not be called.
-
   RETURN VALUE
-    0
+    Name of time zone as String
 */
-const String *Time_zone_utc::get_name() const {
-  /* Should be never called */
-  DBUG_ASSERT(0);
-  return nullptr;
-}
```

Comment:
get_int_sort_key_for_item_inline() changes thd->time_zone and restores it.

From the stack trace reported on bug page, Sort_param::make_sortkey calls val_date_temporal().
Why is there an effect by above culprit change to UTC? it is restored to old time zone right?

Reply:
@satya-bodapati here is a completely resolved stack trace:
```
#0  Time_zone_utc::get_name (this=0x55555cfe7240 <tz_UTC>) at /work/ps/src/8.0/sql/tztime.cc:1177
#1  0x0000555558b8fc92 in Sys_var_tz::session_value_ptr (this=0x55555d3b82e0 <Sys_time_zone>, target_thd=0x7fff34001040) at /work/ps/src/8.0/sql/sys_vars.h:2053
#2  0x00005555588d2a78 in sys_var::value_ptr (this=0x55555d3b82e0 <Sys_time_zone>, running_thd=0x7fff34001040, target_thd=0x7fff34001040, type=OPT_SESSION, base=0x7fff34a0b0d8) at /work/ps/src/8.0/sql/set_var.cc:375
#3  0x00005555588d2aeb in sys_var::value_ptr (this=0x55555d3b82e0 <Sys_time_zone>, thd=0x7fff34001040, type=OPT_SESSION, base=0x7fff34a0b0d8) at /work/ps/src/8.0/sql/set_var.cc:380
#4  0x00005555586a9f10 in Item_func_get_system_var::val_str (this=0x7fff34a0aff0, str=0x7fff34a0b0f8) at /work/ps/src/8.0/sql/item_func.cc:6828
#5  0x00005555585f1939 in Item::val_str_ascii (this=0x7fff34a0aff0, str=0x7ffff02739b0) at /work/ps/src/8.0/sql/item.cc:234
#6  0x000055555874efed in Item_func_convert_tz::get_date (this=0x7fff34a0ada0, ltime=0x7ffff0273a10, fuzzy_date=1) at /work/ps/src/8.0/sql/item_timefunc.cc:2038
#7  0x000055555874abc9 in Item_datetime_func::val_date_temporal (this=0x7fff34a0ada0) at /work/ps/src/8.0/sql/item_timefunc.cc:926
#8  0x0000555558db889a in (anonymous namespace)::get_int_sort_key_for_item_inline (item=0x7fff34a0ada0) at /work/ps/src/8.0/sql/filesort.cc:1213
#9  0x0000555558db8de4 in (anonymous namespace)::make_sortkey_from_item (item=0x7fff34a0ada0, result_type=INT_RESULT, dst_length=..., tmp_buffer=0x7fff34a0c158, to=0x7fff34a3a961 "Ufw\210\231\252\273/row/row0sel.cc", to_end=0x7fff34a40200 "@\245\\\a", maybe_null=0x7ffff0273c4a, hash=0x7ffff0273c58) at /work/ps/src/8.0/sql/filesort.cc:1305
#10 0x0000555558db94c4 in Sort_param::make_sortkey (this=0x7fff34a0c100, dst=..., ref_pos=0x7fff34a02d90 "", longest_addon_so_far=0x7ffff0273df0) at /work/ps/src/8.0/sql/filesort.cc:1408
#11 0x0000555558db7aaf in alloc_and_make_sortkey (param=0x7fff34a0c100, fs_info=0x7fff34a0c2c8, ref_pos=0x7fff34a02d90 "", key_length=0x7ffff0273e00, longest_addons=0x7ffff0273df0) at /work/ps/src/8.0/sql/filesort.cc:843
#12 0x0000555558db7e68 in read_all_rows (thd=0x7fff34001040, param=0x7fff34a0c100, fs_info=0x7fff34a0c2c8, chunk_file=0x7ffff0274210, tempfile=0x7ffff02740e0, pq=0x0, source_iterator=0x7fff34a0c238, found_rows=0x7ffff02743e8, longest_key=0x7ffff0273f28, longest_addons=0x7ffff0273f30) at /work/ps/src/8.0/sql/filesort.cc:990
#13 0x0000555558db64a2 in filesort (thd=0x7fff34001040, filesort=0x7fff34a0c0d0, source_iterator=0x7fff34a0c238, fs_info=0x7fff34a0c2c8, sort_result=0x7fff34a0c350, found_rows=0x7ffff02743e8) at /work/ps/src/8.0/sql/filesort.cc:506
#14 0x00005555588dae35 in SortingIterator::DoSort (this=0x7fff34a0c280) at /work/ps/src/8.0/sql/sorting_iterator.cc:605
#15 0x00005555588da767 in SortingIterator::Init (this=0x7fff34a0c280) at /work/ps/src/8.0/sql/sorting_iterator.cc:514
#16 0x0000555558b542bc in SELECT_LEX_UNIT::ExecuteIteratorQuery (this=0x7fff34a08b78, thd=0x7fff34001040) at /work/ps/src/8.0/sql/sql_union.cc:1166
#17 0x0000555558b546b2 in SELECT_LEX_UNIT::execute (this=0x7fff34a08b78, thd=0x7fff34001040) at /work/ps/src/8.0/sql/sql_union.cc:1236
#18 0x0000555558aa4ce2 in Sql_cmd_dml::execute_inner (this=0x7fff34a0b170, thd=0x7fff34001040) at /work/ps/src/8.0/sql/sql_select.cc:946
#19 0x0000555558aa418b in Sql_cmd_dml::execute (this=0x7fff34a0b170, thd=0x7fff34001040) at /work/ps/src/8.0/sql/sql_select.cc:726
#20 0x0000555558a25925 in mysql_execute_command (thd=0x7fff34001040, first_level=true) at /work/ps/src/8.0/sql/sql_parse.cc:4706
#21 0x0000555558a28543 in mysql_parse (thd=0x7fff34001040, parser_state=0x7ffff0275b90, update_userstat=false) at /work/ps/src/8.0/sql/sql_parse.cc:5530
#22 0x0000555558a1c856 in dispatch_command (thd=0x7fff34001040, com_data=0x7ffff0276c40, command=COM_QUERY) at /work/ps/src/8.0/sql/sql_parse.cc:1826
#23 0x0000555558a1ac0f in do_command (thd=0x7fff34001040) at /work/ps/src/8.0/sql/sql_parse.cc:1303
#24 0x0000555558c15112 in handle_connection (arg=0x55555dc6b4d0) at /work/ps/src/8.0/sql/conn_handler/connection_handler_per_thread.cc:308
#25 0x000055555a63f2d7 in pfs_spawn_thread (arg=0x55555f25f0d0) at /work/ps/src/8.0/storage/perfschema/pfs.cc:2855
#26 0x00007ffff77e0609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#27 0x00007ffff7705103 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
```
`val_date_temporal` calls `get_date` (frame 6), that function will find from and to TZ using timezone name as parameter, we are now converting to UTC, which will end-up in `Time_zone_utc::get_name ` . So we actually need to access get_name from utc object before we restore back to original session tz.

Reply:
Thanks, with resolved stack trace it is clear now